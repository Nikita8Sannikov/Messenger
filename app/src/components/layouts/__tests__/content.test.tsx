import { render, screen } from "@testing-library/react";
import ContentLayout from "@components/layouts/content";

describe("ContentLayout", () => {
  it("renders children inside BoxComponent", () => {
    const childrenContent = <div>Test content</div>;

    render(<ContentLayout>{childrenContent}</ContentLayout>);

    // Проверяем, что содержимое дочерних элементов отобразилось
    expect(screen.getByText("Test content")).toBeInTheDocument();
  });

  it("passes className .content to BoxComponent", () => {
    render(<ContentLayout>{<div>Test content</div>}</ContentLayout>);

    // Проверяем, что класс передается в BoxComponent
    const box = screen.getByText("Test content").parentElement;
    expect(box).toHaveClass("content");
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<ContentLayout>{<div>Test content</div>}</ContentLayout>);

    // Снимаем снапшот компонента
    expect(asFragment()).toMatchSnapshot();
  });
});
