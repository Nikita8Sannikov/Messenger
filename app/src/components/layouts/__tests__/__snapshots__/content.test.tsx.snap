// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentLayoutComponent matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="main-content"
  >
    <div>
      Test content
    </div>
  </div>
</DocumentFragment>
`;
